src/cutadapt/adapters.py

```diff
@@ -698,6 +734,14 @@ def _aligner(self) -> Aligner:
         self.sequence = self.sequence[::-1]
         return aligner
 
+    def _kmer_finder(self):
+        self.sequence = self.sequence[::-1]
+        kmer_finder = self._make_kmer_finder(
+            front_adapter=True, back_adapter=self._force_anywhere
```

Comment:
I believe this needs to be adjusted to fix the CI failure, see 34defe5bd83cf01dd948763478cd57f6cffd4fc4

Reply:
Thanks. I was worried I had created some weird hardware-implementation dependent bug with the shift-OR implementation. (You never know with C...)
I can reproduce the issue now, so I should also be able to fix it.

Reply:
Done. All tests pass. Thanks again!